src/boot/bootctl-uki.c

```diff
@@ -143,21 +144,49 @@ static int read_pe_section(FILE *uki, const struct PeSectionHeader *section,
         return 0;
 }
 
+static int inspect_osrel(char *osrel, size_t osrel_size) {
```

Comment:
it's one of those case where i'd probably pass const and cast away the const.

i mean, you pass "r" to fmemopen(), i.e. you open the buffer in read-only mode with it. The only reason why fmemopen() doesn't put "const" on its first argument is simply because the C language cannot express that "hey, if the second arg is `r` then the first arg is actually const". Conceptually though the memory never should be touched throughout inspect_osrel(), hence I think it would make sense to communicate that in the function prototype and just do the necessary casts internally, and hide them away inside the function.

but not a biggie